services/QuillLMS/client/app/bundles/Staff/components/evidence/healthDashboards/promptHealthRow.tsx

```diff
@@ -0,0 +1,157 @@
+    },
+    {
+      Header: 'Final Attempt Optimal',
+      accessor: "final_attempt_optimal",
+      key: "final_attempt_optimal",
+      minWidth: 70,
+      Cell: ({row}) => row.original.final_attempt_optimal && <span className={row.original.final_attempt_optimal < 75 ? "poor-health" : "" + " name"}>{row.original.final_attempt_optimal}%</span>
+    },
+    {
+      Header: 'Average Attempts',
+      accessor: "avg_attempts",
+      key: "avg_attempts",
+      minWidth: 70,
+    },
+    {
+      Header: 'Confid- ence',
+      accessor: "confidence",
+      key: "confidence",
+      minWidth: 70,
+      Cell: ({row}) => row.original.confidence && <span className={row.original.confidence < 0.9 ? "poor-health" : "" + " name"}>{row.original.confidence}%</span>
+    },
+    {
+      Header: '% AutoML Consecut- ive Repeated',
+      accessor: "percent_automl_consecutive_repeated",
+      key: "percent_automl_consecutive_repeated",
+      minWidth: 70,
+      Cell: ({row}) => row.original.percent_automl_consecutive_repeated && <span className={row.original.percent_automl_consecutive_repeated > 30 ? "poor-health" : "" + " name"}>{row.original.percent_automl_consecutive_repeated}%</span>
+    },
+    {
+      Header: '% AutoML',
+      accessor: "percent_automl",
+      key: "percent_automl",
+      minWidth: 70,
+    },
+    {
+      Header: '% Plagiar- ism',
+      accessor: "percent_plagiarism",
+      key: "percent_plagiarism",
+      minWidth: 70,
+    },
+    {
+      Header: '% Opinion',
+      accessor: "percent_opinion",
+      key: "percent_opinion",
+      minWidth: 70,
+    },
+    {
+      Header: '% Grammar',
+      accessor: "percent_grammar",
+      key: "percent_grammar",
+      minWidth: 70,
+    },
+    {
+      Header: '% Spelling',
+      accessor: "percent_spelling",
+      key: "percent_spelling",
+      minWidth: 70,
+    },
+    {
+      Header: 'Avg Time Spent - Prompt',
+      accessor: "avg_time_spent_per_prompt",
+      key: "avg_time_spent_per_prompt",
+      minWidth: 70,
+      Cell: ({row}) => secondsToHumanReadableTime(row.original.avg_time_spent_per_prompt)
+    },
+    {
+      Header: 'Poor Health Flag',
+      accessor: "poor_health_flag",
+      key: "poorHealthFlag",
+      minWidth: 70,
+      Cell: ({row}) => row.original.poor_health_flag === true && <span className="poor-health-flag">X</span>,
+    }
+  ];
+
+  const renderTable = () => {
+    return tableOrEmptyMessage()
+  }
+
+  const tableOrEmptyMessage = () => {
+    let tableOrEmptyMessage
+
+    if (data.length) {
+      tableOrEmptyMessage = (<ReactTable
+        className='prompt-health-dropdown-table'
+        collapseOnDataChange={false}
+        columns={dataTableFields}
+        data={data}
+        defaultPageSize={data.length}
+        loading={false}
+        pages={1}
+      />)
+    } else {
+      tableOrEmptyMessage = NO_DATA_FOUND_MESSAGE
+    }
+    return (
+      <div>
+        {tableOrEmptyMessage}
+      </div>
+    )
+  }
```

Comment:
Just a small stylistic suggestion, but you could utilize an early return here and get rid of the variable (and just a note for good accessibility practice, I'd also pass the no data message in a p tag since it's a text element rather than visual):

```
  const renderTableOrEmptyMessage = () => {

    if (data.length) {
      return (<ReactTable
        className='prompt-health-dropdown-table'
        collapseOnDataChange={false}
        columns={dataTableFields}
        data={data}
        defaultPageSize={data.length}
        loading={false}
        pages={1}
      />)
    }
    return <p>NO_DATA_FOUND_MESSAGE</p>
  }
  ```
  
  and then you can remove the `renderTable` function and just render this in the return below